### ISSUE TITLE
Potentially false validation errors with VK_IMAGE_CREATE_EXTENDED_USAGE_BIT_KHR

### ISSUE THREAD

**Issue by pppessi** [KhronosGroup/Vulkan-LoaderAndValidationLayers#2464]
Running `dEQP-VK.image.texel_view_compatible.*` produces a bunch of validation errors. Errors from invalid pipeline stage flags are legitimate but the rest seem rather suspicious.

Example errors from `dEQP-VK.image.texel_view_compatible.compute.basic.image_load.bc1_rgb_unorm_block.r16g16b16a16_unorm`

> ERROR: vkCreateImage: usage bit VK_IMAGE_USAGE_STORAGE_BIT is not supported for format VK_FORMAT_BC1_RGB_UNORM_BLOCK with tiling VK_IMAGE_TILING_OPTIMAL. The spec valid usage text states 'If tiling is VK_IMAGE_TILING_OPTIMAL, and VkFormatProperties::optimalTilingFeatures (as returned by vkGetPhysicalDeviceFormatProperties with the same value of format) does not include VK_FORMAT_FEATURE_STORAGE_IMAGE_BIT, usage must not contain VK_IMAGE_USAGE_STORAGE_BIT' (https://www.khronos.org/registry/vulkan/specs/1.0/html/vkspec.html#VUID-VkImageCreateInfo-tiling-00984) (code 0x09e007b0 from IMAGE at Unknown:726)

> ERROR: vkCreateImage: The combination of format, type, tiling, usage and flags supplied in the VkImageCreateInfo struct is reported by vkGetPhysicalDeviceImageFormatProperties() as unsupported. The spec valid usage text states 'The combination of format, imageType, tiling, usage, and flags must be supported, as indicated by a VK_SUCCESS return value from vkGetPhysicalDeviceImageFormatProperties invoked with the same values passed to the corresponding parameters.' (https://www.khronos.org/registry/vulkan/specs/1.0/html/vkspec.html#VUID-VkImageCreateInfo-format-00940) (code 0x09e00758 from IMAGE at Unknown:771)

These errors are produced when creating a `VK_FORMAT_BC1_RGB_UNORM_BLOCK` image with `(VK_IMAGE_USAGE_TRANSFER_SRC_BIT | VK_IMAGE_USAGE_TRANSFER_DST_BIT | VK_IMAGE_USAGE_SAMPLED_BIT | VK_IMAGE_USAGE_STORAGE_BIT)` usage flags and `(VK_IMAGE_CREATE_MUTABLE_FORMAT_BIT | VK_IMAGE_CREATE_BLOCK_TEXEL_VIEW_COMPATIBLE_BIT_KHR | VK_IMAGE_CREATE_EXTENDED_USAGE_BIT_KHR)` create flags.

The validation layer skips checks for `VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT` and `VK_IMAGE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT` usage if `VK_IMAGE_CREATE_EXTENDED_USAGE_BIT_KHR` is included in `VkImageCreateFlags`, but the checks for `VK_IMAGE_USAGE_SAMPLED_BIT` and `VK_IMAGE_USAGE_STORAGE_BIT` are always performed. Until a proper support for extended usage validation is implemented, these checks should probably be skipped as well.

> ERROR: vkCreateImageView() pCreateInfo->format VK_FORMAT_BC1_RGB_UNORM_BLOCK cannot be used with an image having the VK_IMAGE_TILING_OPTIMAL and VK_IMAGE_USAGE_STORAGE_BIT flags set. The spec valid usage text states 'If image was created with VK_IMAGE_TILING_OPTIMAL and usage contains VK_IMAGE_USAGE_STORAGE_BIT, format must be supported for storage images, as specified by the VK_FORMAT_FEATURE_STORAGE_IMAGE_BIT flag in VkFormatProperties::optimalTilingFeatures returned by vkGetPhysicalDeviceFormatProperties with the same value of format' (https://www.khronos.org/registry/vulkan/specs/1.0/html/vkspec.html#VUID-VkImageViewCreateInfo-image-01014) (code 0x0ac007ec from IMAGE at Unknown:3514)

This seems to happen because the validation layer checks the implicit usage flags inherited from image creation time instead of usage flags from `VkImageViewUsageCreateInfoKHR`. The latter should override the former if `VkImageViewUsageCreateInfoKHR` is chained to `VkImageViewCreateInfo`.

TLDR:
- Image usage flags in `vkCreateImage` are not validated correctly if `VK_IMAGE_CREATE_EXTENDED_USAGE_BIT_KHR` is used
- Image usage flags in `vkCreateImageView` are not validated correctly if `VkImageViewUsageCreateInfoKHR` is used

**Comment by mark-lunarg** [comment]
@pppessi, thanks for the report.  We have a TODO in place at the moment and will bump the priority on this issue.

**Comment by daveh-lunarg** [comment]
The first will be easy to avoid with the existing work-around which avoids _probably-_ false positives when the EXTENED_USAGE bit is set. The full solution will require an exhaustive search of all compatible formats, unless we can get clever about caching the app's queries.

The second is harder to triage, because the ```VkImageViewUsageCreateInfoKHR``` is only allowed to include a subset of the image's creation-time usage bits so can only make the check __more__ restrictive, so the fact we're not traversing the chain currently is probably not the root cause of the false positive.

**Comment by daveh-lunarg** [comment]
The linked PR will avoid false positives mentioned in the first part of the issue, and adds support for a chained VkImageViewUsageCreateInfo struct mentioned in the second part.  It's not clear to me whether this actually fixes the errors of the second part.
